main/api/src/mill/api/PathRef.scala

```diff
@@ -45,9 +46,21 @@ object PathRef {
               val value = (attrs.mtime, attrs.size).hashCode()
               updateWithInt(value)
             } else if (jnio.Files.isReadable(path.toNIO)) {
-              val is = os.read.inputStream(path)
-              StreamSupport.stream(is, digestOut)
-              is.close()
+              val is =
+                try Some(os.read.inputStream(path))
+                catch {
+                  case _: jnio.FileSystemException =>
+                    // This is known to happen, when we try to digest a socket file.
```

Comment:
Maybe a Debug log of the exception would be nice here.
At the same time you probably don't have access to the logger here, right?

Reply:
That was exactly my initial thought. At least I tried to minimize the try-catch to the creation of the input stream, so we don't swallow other errors while reading.